### PR TITLE
chore(release): prepare 24.05.0

### PR DESCRIPTION
**Pull request by tuntoja**
## Description

* bump all components to 24.05.0
* update containers

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] 24.05.x (master)

## Summary by CodeRabbit

- **New Features**
  - Updated Centreon modules and widgets to version `24.05.0`.
  - Introduced conditional logic in Dockerfiles to select repository URLs based on `RELEASE_CLOUD` status.

- **Documentation**
  - Updated CHANGELOG for Centreon HA to include new version `24.05.0`.

- **Refactor**
  - Modified GitHub Actions workflows to enhance branch filtering logic and set release parameters based on branch names.

- **Chores**
  - Incremented major version numbers across various configuration files and environment settings to align with new release `24.05.0`.

### CODE REVIEW

File: centreon/www/install/insertBaseConf.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@
 -- Insert version
 --
 
-INSERT INTO `informations` (`key` ,`value`) VALUES ('version', '24.04.0');
+INSERT INTO `informations` (`key` ,`value`) VALUES ('version', '24.05.0');
```
[the remark this answers is not in the document]
Comment (tuntoja) [reply centreon/www/install/insertBaseConf.sql:5]: This should be processed in a different PR as the PR #3855 only focuses on preparing the upcoming version, not on ensuring database data integrity.

File: centreon/tests/e2e/features/Platform-upgrade-update/common.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ const installCentreon = (version: string): Cypress.Chainable => {
         `echo 'date.timezone = Europe/Paris' > /etc/php.d/centreon.ini`,
         `/etc/init.d/mysql start`,
         `mkdir -p /run/php-fpm`,
-        `systemctl start php-fpm`,
+        `systemctl start php-fpm || systemctl restart php-fpm`,
         `systemctl start httpd`,
```
Comment (sc979) [inline centreon/tests/e2e/features/Platform-upgrade-update/common.ts:104]:
```suggestion
        `systemctl start httpd || systemctl restart httpd`,
```

File: .github/workflows/get-version.yml [hunk 3]
```diff
@@ -133,6 +133,11 @@ jobs:
               echo "release_cloud=1" >> $GITHUB_OUTPUT
               echo "release_type=$GITHUB_RELEASE_TYPE" >> $GITHUB_OUTPUT
               ;;
+            [2-9][0-9].[0-9][0-9].x)
```
Comment (sc979) [inline .github/workflows/get-version.yml:136]: to check

File: .github/workflows/ha.yml [hunk 4]
```diff
@@ -31,135 +29,135 @@ jobs:
...
+#       major_version: ${{ needs.get-version.outputs.major_version }}
+#       minor_version: ${{ needs.get-version.outputs.minor_version }}
+#       release: ${{ needs.get-version.outputs.release }}
+#       commit_hash: ${{ github.sha }}
+#       cache_key: ${{ github.sha }}-${{ github.run_id }}-${{ matrix.package_extension }}-${{ matrix.distrib }}
+#       distrib: ${{ matrix.distrib }}
+#     secrets:
+#       registry_username: ${{ secrets.DOCKER_REGISTRY_ID }}
+#       registry_password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+
```
[the remark this answers is not in the document]
Comment (tuntoja) [reply .github/workflows/ha.yml:69]: disabling ha.yml workflow is intended to be permanent on develop and master branches, or any children release or hotfix branch coming from develop or master

**Review by jeremyjaouen — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**